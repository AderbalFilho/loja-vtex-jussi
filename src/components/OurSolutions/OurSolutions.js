import React from 'react';
import PropTypes from 'prop-types';

import OurSolutionsCard from '@components/OurSolutionsCard';

import { JSection, JTitle } from '@ui';

import './OurSolutions.scss';

function OurSolutions({ solutions }) {
  return (
    <JSection classes="our-solutions">
      <div className="our-solutions__container">
        <div className="our-solutions__titles">
          <JTitle tag="h2" color="green">
            &#47;&#47;
          </JTitle>
          <JTitle tag="h2">Nossas soluções</JTitle>
        </div>
        <div className="our-solutions__cards">
          {solutions.map((solution) => {
            return <OurSolutionsCard key={solution.name} solution={solution} />;
          })}
        </div>
      </div>
    </JSection>
  );
}

OurSolutions.propTypes = {
  solutions: PropTypes.arrayOf(
    PropTypes.shape({
      description: PropTypes.string.isRequired,
      features: PropTypes.arrayOf(PropTypes.string).isRequired,
      img: PropTypes.string.isRequired,
      imgAlt: PropTypes.string.isRequired,
      link: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default OurSolutions;
