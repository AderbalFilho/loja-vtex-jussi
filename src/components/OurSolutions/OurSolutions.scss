.our-solutions {
  padding: 56px 0;
}

.our-solutions__container {
  margin: 0 auto;
  max-width: 1024px;
  width: 100%;
}

.our-solutions__titles {
  display: flex;

  h2 {
    margin: 0 0 24px 0;

    &:first-child {
      margin-right: 4px;
    }
  }
}

.our-solutions__cards {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;

  @media (min-width: 768px) {
    justify-content: space-between;
  }
}
