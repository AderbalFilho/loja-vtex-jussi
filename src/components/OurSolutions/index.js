import OurSolutions from './OurSolutions';

export default OurSolutions;
