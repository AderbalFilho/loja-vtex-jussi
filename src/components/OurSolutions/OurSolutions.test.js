import React from 'react';
import TestRenderer from 'react-test-renderer';

import OurSolutions from './OurSolutions';
import { SOLUTIONS } from '@constants';

describe('OurSolutions', () => {
  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(
      <OurSolutions solutions={SOLUTIONS} />
    );
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });
});
