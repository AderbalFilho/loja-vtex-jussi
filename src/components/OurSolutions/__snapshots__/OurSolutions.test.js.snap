// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OurSolutions matches snapshot 1`] = `
<section
  className="j-section j-section--white our-solutions"
>
  <div
    className="our-solutions__container"
  >
    <div
      className="our-solutions__titles"
    >
      <h2
        className="j-title j-title--green j-title--large"
      >
        //
      </h2>
      <h2
        className="j-title j-title--black j-title--large"
      >
        Nossas soluções
      </h2>
    </div>
    <div
      className="our-solutions__cards"
    >
      <div
        className="j-card j-card--rad-normal j-card--size-normal our-solutions-card"
      >
        <img
          alt="P1"
          className="our-solutions-card__img"
          src="https://static.openfoodfacts.org/images/products/073/762/806/4502/ingredients_en.10.400.jpg"
        />
        <div>
          <p
            className="j-text j-text--black j-text--normal our-solutions-card__name"
          >
            Nome do Produto #1
          </p>
          <p
            className="j-text j-text--pink j-text--small our-solutions-card__description"
          >
            Descrição do produto #1
          </p>
          <ul
            className="our-solutions-card__list"
          >
            <li>
              <p
                className="j-text j-text--dark-grey j-text--small"
              >
                Feature 1
              </p>
            </li>
            <li>
              <p
                className="j-text j-text--dark-grey j-text--small"
              >
                Feature 2
              </p>
            </li>
            <li>
              <p
                className="j-text j-text--dark-grey j-text--small"
              >
                Feature 3
              </p>
            </li>
          </ul>
          <a
            className="j-button j-button--back-green j-button--border-green j-button--black j-button--rad-normal j-button--link our-solutions-card__button"
            href=""
            rel="noreferrer noopener"
            target="_blank"
          >
            Ver solução
          </a>
        </div>
      </div>
      <div
        className="j-card j-card--rad-normal j-card--size-normal our-solutions-card"
      >
        <img
          alt="P2"
          className="our-solutions-card__img"
          src="https://static.openfoodfacts.org/images/products/073/762/806/4502/ingredients_en.10.400.jpg"
        />
        <div>
          <p
            className="j-text j-text--black j-text--normal our-solutions-card__name"
          >
            Nome do Produto #2
          </p>
          <p
            className="j-text j-text--pink j-text--small our-solutions-card__description"
          >
            Descrição do produto #2
          </p>
          <ul
            className="our-solutions-card__list"
          >
            <li>
              <p
                className="j-text j-text--dark-grey j-text--small"
              >
                Feature 1
              </p>
            </li>
            <li>
              <p
                className="j-text j-text--dark-grey j-text--small"
              >
                Feature 2
              </p>
            </li>
            <li>
              <p
                className="j-text j-text--dark-grey j-text--small"
              >
                Feature 3
              </p>
            </li>
          </ul>
          <a
            className="j-button j-button--back-green j-button--border-green j-button--black j-button--rad-normal j-button--link our-solutions-card__button"
            href=""
            rel="noreferrer noopener"
            target="_blank"
          >
            Ver solução
          </a>
        </div>
      </div>
      <div
        className="j-card j-card--rad-normal j-card--size-normal our-solutions-card"
      >
        <img
          alt="P3"
          className="our-solutions-card__img"
          src="https://static.openfoodfacts.org/images/products/073/762/806/4502/ingredients_en.10.400.jpg"
        />
        <div>
          <p
            className="j-text j-text--black j-text--normal our-solutions-card__name"
          >
            Nome do Produto #3
          </p>
          <p
            className="j-text j-text--pink j-text--small our-solutions-card__description"
          >
            Descrição do produto #3
          </p>
          <ul
            className="our-solutions-card__list"
          >
            <li>
              <p
                className="j-text j-text--dark-grey j-text--small"
              >
                Feature 1
              </p>
            </li>
            <li>
              <p
                className="j-text j-text--dark-grey j-text--small"
              >
                Feature 2
              </p>
            </li>
            <li>
              <p
                className="j-text j-text--dark-grey j-text--small"
              >
                Feature 3
              </p>
            </li>
          </ul>
          <a
            className="j-button j-button--back-green j-button--border-green j-button--black j-button--rad-normal j-button--link our-solutions-card__button"
            href=""
            rel="noreferrer noopener"
            target="_blank"
          >
            Ver solução
          </a>
        </div>
      </div>
      <div
        className="j-card j-card--rad-normal j-card--size-normal our-solutions-card"
      >
        <img
          alt="P4"
          className="our-solutions-card__img"
          src="https://static.openfoodfacts.org/images/products/073/762/806/4502/ingredients_en.10.400.jpg"
        />
        <div>
          <p
            className="j-text j-text--black j-text--normal our-solutions-card__name"
          >
            Nome do Produto #4
          </p>
          <p
            className="j-text j-text--pink j-text--small our-solutions-card__description"
          >
            Descrição do produto #4
          </p>
          <ul
            className="our-solutions-card__list"
          >
            <li>
              <p
                className="j-text j-text--dark-grey j-text--small"
              >
                Feature 1
              </p>
            </li>
            <li>
              <p
                className="j-text j-text--dark-grey j-text--small"
              >
                Feature 2
              </p>
            </li>
            <li>
              <p
                className="j-text j-text--dark-grey j-text--small"
              >
                Feature 3
              </p>
            </li>
          </ul>
          <a
            className="j-button j-button--back-green j-button--border-green j-button--black j-button--rad-normal j-button--link our-solutions-card__button"
            href=""
            rel="noreferrer noopener"
            target="_blank"
          >
            Ver solução
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;
